### PR TITLE
[CI/tests] Mark regtest tests instead of skipping them

### PR DESCRIPTION
**Pull request by merc1er**
The regtest tests were skipped as they require a local node and a Bitcoin.com type REST API to run locally. This decreases code coverage below 90%.

Instead of skipping them, I am flagging them in this PR with a `--regtest` marker. By default, the regtest tests are not run, but if `python run_test.py --regtest` or `pytest --regtest` is used, it will run them using default ports to connect to the local node.

### CODE REVIEW

File: run_tests.py [hunk 1]
```diff
@@ -1,16 +1,22 @@
...
 
 
 def main():
     os.chdir(os.path.dirname(os.path.abspath(__file__)))
 
-    subprocess.call(['coverage', 'run', '-m', 'pytest', '-rw'])
+    if '--regtest' in sys.argv:
+        subprocess.call(['coverage', 'run', '-m', 'pytest', '--regtest', '-rw'])
+    else:
+        subprocess.call(
```
Comment (teran-mckinney) [inline run_tests.py:12]: If this errors, will subprocess.call raise an exception?

**Comment by teran-mckinney** [comment]
Looking good but we're still a hair below 90%. Any way we can get over that or should we bump the target to 89%?

**Comment by merc1er** [comment]
OK @teran-mckinney I added a couple more tests to bring coverage to exactly 90% for now 😅

We should probably focus on bringing that number even higher in a later PR.

File: run_tests.py (hunk 1 above)
Comment (teran-mckinney) [reply run_tests.py:12]: In the future, we should probably do `subprocess.run(check=True)`

https://docs.python.org/3/library/subprocess.html#subprocess.run

**Comment by teran-mckinney** [comment]
Nice work!

**Pull request merged**